Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAMSIConfigurationState.ps1

```diff
@@ -0,0 +1,88 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. $PSScriptRoot\..\..\Helpers\Invoke-CatchActions.ps1
+Function Get-ExchangeAMSIConfigurationState {
+    [CmdletBinding()]
+    param ()
+
+    begin {
+        Function Get-AMSIStatusFlag {
+            [CmdletBinding()]
+            [OutputType([bool])]
+            param (
+                [Parameter(Mandatory = $true)]
+                [object]$AMSIParameters
+            )
+
+            try {
+                Switch ($AMSIParameters.Split("=")[1]) {
+                    ("False") { return $false }
+                    ("True") { return $true }
+                    Default { return $null }
+                }
+            } catch {
+                throw
+            }
+        }
+
+        Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+        $amsiState = "Unknown"
+        $amsiOrgWideSetting = $true
+        $amsiConfigurationQuerySuccessful = $false
+    } process {
+        try {
+            Write-Verbose "Trying to query AMSI configuration state"
+            $amsiConfiguration = Get-SettingOverride -ErrorAction Stop | Where-Object { ($_.ComponentName -eq "Cafe") -and ($_.SectionName -eq "HttpRequestFiltering") }
+            $amsiConfigurationQuerySuccessful = $true
+
+            if (($null -ne $amsiConfiguration) -and
+                ($amsiConfiguration.Count -eq 1)) {
+                Write-Verbose "Setting override detected for AMSI configuration"
+                $amsiState = Get-AMSIStatusFlag -AMSIParameters $amsiConfiguration.Parameters -ErrorAction Stop
+                $amsiOrgWideSetting = ($null -eq $amsiConfiguration.Server)
+            } elseif (($null -ne $amsiConfiguration) -and
+                ($amsiConfiguration.Count -gt 1)) {
+                Write-Verbose "$($amsiConfiguration.Count) overrides for the same component and section detected"
+                $amsiMultiConfigObject = @()
+                foreach ($amsiConfig in $amsiConfiguration) {
+                    $amsiState = Get-AMSIStatusFlag -AMSIParameters $amsiConfig.Parameters -ErrorAction Stop
+                    $amsiOrgWideSetting = ($null -eq $amsiConfig.Server)
+                    $amsiConfigTempCustomObject = [PSCustomObject]@{
```

Comment:
@dpaulson45 I think about to put the custom object creation into another method to avoid this duplicate code. Make sense? What do you think?

Reply:
Yeah, this is going to need to change. I don't think it needs to be another method though.

This should be easy to do all within here.

Reply:
Okay, going to improve this part of the code

Reply:
Reduced the complexity of the check. We return a custom object with two properties if no setting override was detected. This is all we need to perform the output.